src/functions/traits.rs

```diff
@@ -1,6 +1,6 @@
-use crate::{error::Error, interpreter::Interpreter, types::Type};
+use crate::{error::ErrorType, interpreter::Interpreter, types::Type};
 
-pub type FResult = Result<Type, Error>;
+pub type FResult = Result<Type, (String, ErrorType)>;
```

Comment:
Why was this changed?

Reply:
I just changed it because it's faster to do that than to replace FResult one-by-one to do the traceback hack.

Reply:
We shouldn't be using hacks. If there is a problem, it needs to be addressed (probably in a different PR) instead of being worked around. The fact that you've changed this means now all of the other functions have to deal with it. A better solution would be to fix the specific problem at hand: to make the line information *optional*. **Do not try to replace the** `Error` **type.**